tests/test_remove_mon_from_cluster.py

```diff
@@ -0,0 +1,126 @@
+"""
+A Testcase to remove one or more mons
+when I/O's are happening.
+
+Polarion-ID- OCS-355
+
+"""
+
+import logging
+import re
+import random
+from ocs import ocp, constants
+from resources import pod
+from ocsci.testlib import tier4, ManageTest
+from ocsci.config import ENV_DATA
+from ocs.exceptions import CephHealthException
+from utility.utils import run_cmd
+
+
+log = logging.getLogger(__name__)
+
+DEP = ocp.OCP(
+    kind=constants.DEPLOYMENT, namespace=ENV_DATA['cluster_namespace']
+)
+POD = ocp.OCP(
+    kind=constants.POD, namespace=ENV_DATA['cluster_namespace']
+)
+
+
+def get_mons_from_cluster():
+    """
+    Getting the list of active mons in the cluster
+    if available = 1 then pod is active or else its inactive/
+    does not exist.
+    """
+
+    ret = DEP.get(resource_name='', out_yaml_format=False,
+                  selector='app=rook-ceph-mon')
+    available_mon = re.findall(r'[\w-]+mon-+[\w-]', ret)
+    return available_mon
+
+
+def verify_mons_are_present():
+    """
+    To verify if there are more than 1 mon
+    present in the cluster
+    """
+    return bool(len(get_mons_from_cluster()))
+
+
+def remove_mon_from_cluster(mon):
+    """
+    Removing the mon pod from deployment
+    """
+    assert DEP.delete(resource_name=mon)
+
+
+def verify_mon_pod_up():
+    ret = POD.wait_for_resource(
+        condition=constants.STATUS_RUNNING, selector='app=rook-ceph-mon',
+        resource_count=5, timeout=700)
+    return ret
+
+
+def run_io():
+    """
+    Run io on the mount point
+    """
+
+    cmd = 'ceph osd lspools'
+    tools_pod = pod.get_ceph_tools_pod()
+    out = tools_pod.exec_ceph_cmd(ceph_cmd=cmd)
+    cmd1 = ('rados bench -p %s 60 write -t 32 -b 409600000 --no-cleanup'
+            % out[0]['poolname'])
+    log.info(cmd1)
+
+
+def ceph_health_check(namespace):
+    """
+    Exec `ceph health` cmd on tools pod to determine health of cluster.
+
+    Args:
+        namespace (str): Namespace of OCS
+            (default: ENV_DATA['cluster_namespace'])
+
+    Raises:
+        CephHealthException: If the ceph health returned is not HEALTH_OK
+        CommandFailed: If the command to retrieve the tools pod name or the
+            command to get ceph health returns a non-zero exit code
+    Returns:
+        boolean: True if HEALTH_OK
+
+    """
+    run_cmd(
+        f"oc wait --for condition=ready pod "
+        f"-l app=rook-ceph-tools "
+        f"-n {namespace} "
+        f"--timeout=120s"
+    )
+    tools_pod = run_cmd(
+        f"oc -n {namespace} get pod -l 'app=rook-ceph-tools' "
+        f"-o jsonpath='{{.items[0].metadata.name}}'"
+    )
+    health = run_cmd(f"oc -n {namespace} exec {tools_pod} ceph health")
```

Comment:
use resources.pod.Pod#exec_ceph_cmd instead

Reply:
$ oc -n openshift-storage exec rook-ceph-tools-76c7d559b6-qmppg ceph health
HEALTH_OK

I directly get the health message if I use this way.
If I want to use exec_ceph_cmd() Then the command would be "ceph status". From the output of "ceph status" I will have to Grep for Health. Hence to avoid complexity I used this way.

Reply:
@Shrivaibavi why can't you run `ceph health` through `exec_ceph_cmd()`?

Reply:
Done, I used exec_ceph_cmd

Reply:
Perhaps this is a good time to move that to a common place, Sure i agree. Pls suggest me a place to move the ceph_health function

Reply:
@Shrivaibavi for now it can probably move to `tests/helpers.py`? This seems like where we are keeping general test utility functions. It might get moved as a part of #207 but this will at least move it to a spot where other tests may utilize.

Reply:
Moved to tests/helpers.py